language/move-lang/src/parser/syntax.rs

```diff
@@ -432,23 +432,34 @@ fn parse_byte_string<'input>(tokens: &mut Lexer<'input>) -> Result<Vec<u8>, Erro
     }
     let start_loc = tokens.start_loc();
     let s = tokens.content();
-    assert!(s.starts_with("x\""));
-    let mut hex_string = String::from(&s[2..s.len() - 1]);
-    if hex_string.len() % 2 != 0 {
-        hex_string.insert(0, '0');
-    }
-    tokens.advance()?;
-    match hex::decode(hex_string.as_str()) {
-        Ok(vec) => Ok(vec),
-        Err(hex::FromHexError::InvalidHexCharacter { c, index }) => {
-            let offset = start_loc + 1 + index;
-            let loc = make_loc(tokens.file_name(), offset, offset);
-            Err(vec![(
-                loc,
-                format!("Invalid hexadecimal character: '{}'", c),
-            )])
+    if s.starts_with("x\"") {
+        assert!(s.starts_with("x\""));
+        let mut hex_string = String::from(&s[2..s.len() - 1]);
+        if hex_string.len() % 2 != 0 {
+            hex_string.insert(0, '0');
+        }
+        tokens.advance()?;
+        match hex::decode(hex_string.as_str()) {
+            Ok(vec) => Ok(vec),
+            Err(hex::FromHexError::InvalidHexCharacter { c, index }) => {
+                let offset = start_loc + 1 + index;
+                let loc = make_loc(tokens.file_name(), offset, offset);
+                Err(vec![(
+                    loc,
+                    format!("Invalid hexadecimal character: '{}'", c),
+                )])
+            }
+            Err(_) => unreachable!("unexpected error parsing hex byte string value"),
         }
-        Err(_) => unreachable!("unexpected error parsing hex byte string value"),
+    } else if s.starts_with("b\"") {
+        let byte_string = String::from(&s[2..s.len() - 1]);
+        tokens.advance()?;
+        Ok(byte_string.into_bytes())
```

Comment:
I don't think this is going to work for escape sequences, I think you will need to manually walk the string, char by char, and translate the byte value

Reply:
Yes, I agree. I will correct it.